core/src/main/scala/org/apache/spark/storage/DiskBlockManager.scala

```diff
@@ -80,6 +77,65 @@ private[spark] class DiskBlockManager(blockManager: BlockManager, conf: SparkCon
     new File(subDir, filename)
   }
 
+  def getFileDefault(filename: String): File = {
+    _getFile(filename, localDirs)
+  }
+  var getFile = getFileDefault _
+
+  private val hierarchyStore = conf.getOption("spark.storage.hierarchyStore")
+  if (hierarchyStore.isDefined) {
+    val HSLayers: Array[(String, Long)] =
+      hierarchyStore.get.trim.split(",").map {
+        s => val x = s.trim.split(" +")
+          (x(0), Utils.byteStringAsGb(x(1)))
+      }
+    val HSLayerDirs: Array[Array[File]] = HSLayers.map(
```

Comment:
I am wondering if we can split the faster devices and the normal devices for backward-compatible.
e.g.
```
spark.storage.hierarchyStore=/mnt/nvm1,/mnt/nvm2,/mnt/ssd1,/mnt/ssd2
spark.local.dir=/mnt/sata1/,/mnt/sata2,/mnt/stat3
```